src/classlibnative/bcltype/system.cpp

```diff
@@ -41,14 +41,35 @@ typedef BOOL (*pfnGetPhoneVersion)(LPOSVERSIONINFO lpVersionInformation);
 pfnGetPhoneVersion g_pfnGetPhoneVersion = NULL;
 #endif
 
+typedef void(WINAPI *pfnGetSystemTimeAsFileTime)(LPFILETIME lpSystemTimeAsFileTime);
+pfnGetSystemTimeAsFileTime g_pfnGetSystemTimeAsFileTime = NULL;
+Volatile<bool> g_fGetSystemTimeAsFileTimeInitialized = false;
 
 FCIMPL0(INT64, SystemNative::__GetSystemTimeAsFileTime)
 {
     FCALL_CONTRACT;
 
-    INT64 timestamp;
+	if (!g_fGetSystemTimeAsFileTimeInitialized)
```

Comment:
It will save a few cycles to get rid volatile check out and move the initialization into separate method. You can do something like this:

```
extern pfnGetSystemTimeAsFileTime g_pfnGetSystemTimeAsFileTime;

void WINAPI InitializeGetSystemTimeAsFileTime(LPFILETIME lpSystemTimeAsFileTime)
{
    HMODULE hKernel32 = WszLoadLibrary(W("kernel32.dll")); 
    if (hKernel32 != NULL)
    {
        g_pfnGetSystemTimeAsFileTime = (pfnGetSystemTimeAsFileTime)GetProcAddress(hKernel32, "GetSystemTimePreciseAsFileTime");
    }
    else
    {
        g_pfnGetSystemTimeAsFileTime = &::GetSystemTimeAsFileTime;
    }

    g_pfnGetSystemTimeAsFileTime(lpSystemTimeAsFileTime);
}

pfnGetSystemTimeAsFileTime g_pfnGetSystemTimeAsFileTime = InitializeGetSystemTimeAsFileTime(LPFILETIME lpSystemTimeAsFileTime);
```

The hot path can then be just `g_pfnGetSystemTimeAsFileTime((FILETIME*)&timestamp);`